janusgraph-test/src/test/java/org/janusgraph/core/ConfiguredGraphFactoryTest.java

```diff
@@ -0,0 +1,280 @@
+import org.apache.commons.configuration.ConfigurationException;
+
+import com.google.common.collect.Iterators;
+import java.util.Map;
+import java.util.HashMap;
+import java.io.IOException;
+import java.io.File;
+import org.junit.Test;
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.rules.ExpectedException;
+import static org.junit.Assert.*;
+import static org.hamcrest.Matchers.equalTo;
+
+public class ConfiguredGraphFactoryTest {
+    private static final JanusGraphManager gm;
+    private static final ConfigurationManagementGraph configGraphManagement;
+    static {
+        gm = new JanusGraphManager(new Settings());
+        final Map<String, Object> map = new HashMap<String, Object>();
+        map.put(STORAGE_BACKEND.toStringWithoutRoot(), "inmemory");
+        final MapConfiguration config = new MapConfiguration(map);
+        final StandardJanusGraph graph = new StandardJanusGraph(new GraphDatabaseConfiguration(new CommonsConfiguration(config)));
+        configGraphManagement = new ConfigurationManagementGraph(graph);
+    }
+
+    @Rule
+    public ExpectedException thrown = ExpectedException.none();
+
+    @After
+    public void cleanUp() {
+        configGraphManagement.removeTemplateConfiguration();
+    }
+
+    @Test
+    public void shouldGetConfigurationManagementGraphInstance() throws ConfigurationManagementGraphNotEnabledException {
+        final ConfigurationManagementGraph thisInstance = ConfigurationManagementGraph.getInstance();
+        assertNotNull(thisInstance);
+    }
+
+    @Test
+    public void shouldOpenGraphUsingConfiguration() throws InterruptedException, BackendException, Exception {
+        try {
+            final Map<String, Object> map = new HashMap<String, Object>();
+            map.put(STORAGE_BACKEND.toStringWithoutRoot(), "inmemory");
+            map.put(GRAPH_NAME.toStringWithoutRoot(), "graph1");
+            configGraphManagement.createConfiguration(new MapConfiguration(map));
+            final StandardJanusGraph graph = (StandardJanusGraph) ConfiguredGraphFactory.open("graph1");
+            assertNotNull(graph);
+        } finally {
+            configGraphManagement.removeConfiguration("graph1");
+            ConfiguredGraphFactory.close("graph1");
+        }
+    }
+
+    @Test
+    public void graphConfigurationShouldBeWhatWeExpect() throws InterruptedException, BackendException, Exception {
+        try {
+            final Map<String, Object> map = new HashMap<String, Object>();
+            map.put(STORAGE_BACKEND.toStringWithoutRoot(), "inmemory");
+            map.put(GRAPH_NAME.toStringWithoutRoot(), "graph1");
+            configGraphManagement.createConfiguration(new MapConfiguration(map));
+            final StandardJanusGraph graph = (StandardJanusGraph) ConfiguredGraphFactory.open("graph1");
+            assertNotNull(graph);
+            assertEquals(graph.getConfiguration().getConfiguration().get(GRAPH_NAME), "graph1");
```

Comment:
Expected value (constant) is first argument to `assertEquals()`. Same for the next line as well.